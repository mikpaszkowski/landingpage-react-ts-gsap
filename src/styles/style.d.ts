import "styled-components";

interface IPalette {
  footerBackground: string;
  font: string;
  fontLight: string;
  playButton: string;
}

declare module "styled-components" {
  export interface DefaultTheme {
    palette: {
      common: {
        black: string;
        white: string;
        theme: string;
        background: string;
      };
      primary: IPalette;
      secondary: IPalette;
    };
  }
}
