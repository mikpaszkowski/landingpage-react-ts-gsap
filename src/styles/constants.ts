interface Fonts {
  primaryFont: string;
  secondaryFont: string;
}

interface BreakPoints {
  mobileSmall: string;
  mobileMedium: string;
  mobileLarge: string;
  tablet: string;
  tabletLarge: string;
  laptopSmall: string;
  laptopLarge: string;
}

export const fontTypes: Fonts = {
  primaryFont: "'Rozha One', serif",
  secondaryFont: "'Reem Kufi', sans-serif",
};

export const breakPoints: BreakPoints = {
  mobileSmall: "320px",
  mobileMedium: "375px",
  mobileLarge: "525px",
  tablet: "768px",
  tabletLarge: "1110px",
  laptopSmall: "1280px",
  laptopLarge: "1440px",
};

export {};
