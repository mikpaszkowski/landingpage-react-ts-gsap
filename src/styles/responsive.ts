import { breakPoints } from "./styleConstants";

interface DeviceResponsive {
  mobileMedium: string;
  laptopSmall: string;
  laptopLarge: string;
}

export const device: DeviceResponsive = {
  mobileMedium: `(max-width: ${breakPoints.mobileMedium})`,
  laptopSmall: `(max-width: ${breakPoints.laptopSmall})`,
  laptopLarge: `(max-width: ${breakPoints.laptopLarge})`,
};
