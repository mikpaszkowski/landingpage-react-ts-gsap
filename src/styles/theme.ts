import { DefaultTheme } from "styled-components";

export const lightTheme: DefaultTheme = {
  palette: {
    common: {
      black: "#131E07",
      white: "#FFFFFF",
      theme: "#FABD20",
      background: "#FEF4E6",
    },
    primary: {
      footerBackground: "#131E07",
      font: "#131E07",
      fontLight: "#E5E5E5",
      playButton: "#131E07",
    },
    secondary: {
      footerBackground: "#131E07",
      font: "#40462F",
      fontLight: "#7C8176",
      playButton: "#131E07",
    },
  },
};

export const darkTheme: DefaultTheme = {
  palette: {
    common: {
      black: "#131E07",
      white: "#FFFFFF",
      theme: "#FABD20",
      background: "#131E07",
    },
    primary: {
      footerBackground: "#131E07",
      font: "#FFFFFF",
      fontLight: "#E5E5E5",
      playButton: "#131E07",
    },
    secondary: {
      footerBackground: "#131E07",
      font: "#E5E5E5",
      fontLight: "#7C8176",
      playButton: "#131E07",
    },
  },
};
