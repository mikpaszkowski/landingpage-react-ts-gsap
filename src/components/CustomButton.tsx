import * as React from "react";
import styled from "styled-components";
import { colors, fontTypes } from "../styles/styleConstants";

const Button = styled.button`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 0.7rem 3.1rem;
  border: none;
  border-radius: 8px;
  background-color: ${(props: ButtonProps) =>
    props.color ? props.color : colors.primaryTheme};

  & > p {
    font-family: ${fontTypes.secondaryFont};
    font-size: ${(props: ButtonProps) =>
      props.fontsize ? props.fontsize : "2.5rem"};
    font-style: normal;
    font-weight: normal;
    line-height: ${(props: ButtonProps) =>
      props.lineHeight ? props.lineHeight : "3.6rem"};
  }
`;

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  color?: string;
  children?: React.ReactNode;
  lineHeight: string;
  fontsize: string;
}

export const CustomButton = ({ children, ...props }: ButtonProps) => (
  <Button {...props}>
    <p>{children}</p>
  </Button>
);
