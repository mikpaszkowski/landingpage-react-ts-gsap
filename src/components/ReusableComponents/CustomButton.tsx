import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../styles/styleConstants";

const Button = styled.button`
  position: relative;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 0.7rem 3.1rem;
  font-family: ${fontTypes.secondaryFont};
  font-size: ${(props: ButtonProps) =>
    props.fontSize ? props.fontSize : "2.5rem"};
  font-style: normal;
  font-weight: normal;
  line-height: ${(props: ButtonProps) =>
    props.lineHeight ? props.lineHeight : "3.6rem"};
  color: ${({ theme }) => theme.palette.common.black};
  border: none;
  border-radius: ${(props: ButtonProps) =>
    props.borderRadius ? props.borderRadius : "8px"};
  background-color: ${(props: ButtonProps) =>
    props.color ? props.color : ({ theme }) => theme.palette.common.theme};
  cursor: pointer;
  transition: all 0.25s ease-in-out;
  overflow: hidden;
  will-change: transform;

  &::after {
    content: "";
    position: absolute;
    z-index: -1;
    bottom: -100%;
    left: 0;
    height: 100%;
    width: 100%;
    border-radius: ${(props: ButtonProps) =>
      props.borderRadius ? props.borderRadius : "8px"};
    background-color: black;
    color: ${({ theme }) => theme.palette.common.theme};
    transition: all 0.15s cubic-bezier(0.455, 0.03, 0.515, 0.955);
  }

  &:hover {
    box-shadow: 0px 1rem 2rem rgba(0, 0, 0, 0.25);
    color: ${({ theme }) => theme.palette.common.theme};

    &::after {
      transform: translateY(-100%);
    }
  }
`;

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  color?: string;
  name?: string;
  lineHeight?: string;
  fontSize?: string;
  borderRadius?: string;
}

export const CustomButton = ({ name, ...props }: ButtonProps) => (
  <Button {...props}>{name}</Button>
);
