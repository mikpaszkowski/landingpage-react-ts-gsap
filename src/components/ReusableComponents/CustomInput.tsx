import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../styles/styleConstants";
import { device } from "../../styles/responsive";

const CustomInputWrapper = styled.input`
  position: relative;
  width: 35rem;
  height: 5rem;
  background-color: transparent;
  font-family: ${fontTypes.secondaryFont};
  color: ${({ theme }) => theme.palette.secondary.fontLight};
  font-style: normal;
  font-weight: normal;
  font-size: 1.8rem;
  line-height: 2.7rem;
  border: 1px solid ${({ theme }) => theme.palette.secondary.fontLight};
  border-radius: 8px;
  padding: 1.2rem 14rem 1.2rem 1.6rem;

  @media ${device.mobileMedium} {
    width: 100%;
  }
`;

interface CustomInputProps {
  placeholder?: string;
}

export const CustomInput = ({ ...props }: CustomInputProps) => {
  return (
    <div>
      <CustomInputWrapper {...props} />
    </div>
  );
};
