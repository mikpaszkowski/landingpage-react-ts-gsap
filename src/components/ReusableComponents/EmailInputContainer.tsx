import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../styles/styleConstants";
import { CustomInput } from "./CustomInput";
import { CustomButton } from "./CustomButton";
import { device } from "../../styles/responsive";

const EmailInputContainerWrapper = styled.div`
  position: relative;
  & > p {
    font-family: ${fontTypes.secondaryFont};
    font-style: normal;
    font-weight: normal;
    font-size: 1.8rem;
    line-height: 2.7rem;
    margin-bottom: 1.6rem;
    color: ${({ theme }) => theme.palette.secondary.fontLight};
  }
`;

const CustomButtonStyled = styled(CustomButton)`
  position: absolute;
  right: 0.8rem;
  bottom: 0.8rem;
  padding: 0.3rem 1.6rem 0.4rem 1.6rem;
  border-radius: 4px;

  @media ${device.mobileMedium} {
    position: relative;
    width: 100%;
    right: 0;
    bottom: 0;
    font-size: 2rem;
    padding: 1.6rem 8rem 1.6rem 8rem;
    margin: 1.6rem 0 6.3rem 0;
  }
`;

export const EmailInputContainer = () => {
  return (
    <EmailInputContainerWrapper>
      <p>Try Medli for free</p>
      <CustomInput placeholder="Enter your email" />
      <CustomButtonStyled
        fontSize="1.8rem"
        lineHeight="2.7rem"
        borderRadius="4px"
        name="Get started"
      />
    </EmailInputContainerWrapper>
  );
};
