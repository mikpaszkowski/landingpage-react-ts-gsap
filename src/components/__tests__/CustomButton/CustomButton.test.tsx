import { cleanup } from "@testing-library/react";
import "jest-styled-components";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import { CustomButton } from "../../ReusableComponents/CustomButton";
import { ThemeProvider } from "styled-components";
import { lightTheme } from "../../../styles/theme";

afterEach(() => {
  cleanup();
});

test("check render correctly <CustomButton /> component", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={lightTheme}>
        <CustomButton />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe("check CustomButton properties and props", () => {
  test("should have name prop displayed", () => {
    const props = {
      name: "Log in",
    };

    const wrapper = mount(
      <ThemeProvider theme={lightTheme}>
        <CustomButton {...props} />
      </ThemeProvider>
    ).find(CustomButton);

    expect(wrapper.prop("name")).toEqual("Log in");
    expect(wrapper.contains("Log in")).toEqual(true);
  });

  test("should have #000000 - black background-color property", () => {
    const props = {
      color: "#000000",
    };

    const wrapper = mount(
      <ThemeProvider theme={lightTheme}>
        <CustomButton {...props} />
      </ThemeProvider>
    ).find(CustomButton);

    expect(wrapper).toHaveStyleRule("background-color", "#000000");
  });

  test("should have font-size property set to 15rem", () => {
    const props = {
      fontSize: "15rem",
    };

    const wrapper = mount(
      <ThemeProvider theme={lightTheme}>
        <CustomButton {...props} />
      </ThemeProvider>
    ).find(CustomButton);

    expect(wrapper).toHaveStyleRule("font-size", "15rem");
  });

  test("should have line-height property set to 3rem", () => {
    const props = {
      lineHeight: "3rem",
    };

    const wrapper = mount(
      <ThemeProvider theme={lightTheme}>
        <CustomButton {...props} />
      </ThemeProvider>
    ).find(CustomButton);

    expect(wrapper).toHaveStyleRule("line-height", "3rem");
  });

  test("should have border-radius property set to 0.8rem", () => {
    const props = {
      borderRadius: "0.8rem",
    };

    const wrapper = mount(
      <ThemeProvider theme={lightTheme}>
        <CustomButton {...props} />
      </ThemeProvider>
    ).find(CustomButton);

    expect(wrapper).toHaveStyleRule("border-radius", "0.8rem");
  });
});

describe("check props data types", () => {
  test("render component with undefined name props", () => {
    const wrapper = mount(
      <ThemeProvider theme={lightTheme}>
        <CustomButton />
      </ThemeProvider>
    ).find(CustomButton);

    expect(wrapper.prop("name")).toEqual(undefined);
    expect(wrapper.prop("color")).toEqual(undefined);
    expect(wrapper.prop("lineHeight")).toEqual(undefined);
    expect(wrapper.prop("fontSize")).toEqual(undefined);
    expect(wrapper.prop("borderRadius")).toEqual(undefined);
  });

  test("should have name prop as string data type", () => {
    const props = {
      name: "Log in",
    };
    const CustomButtonElement = mount(
      <ThemeProvider theme={lightTheme}>
        <CustomButton {...props} />
      </ThemeProvider>
    ).find(CustomButton);

    expect(typeof CustomButtonElement.prop("name")).toBe("string");
  });

  test("should have color prop as string data type", () => {
    const props = {
      color: "#FFFFFF",
    };
    const wrapper = mount(
      <ThemeProvider theme={lightTheme}>
        <CustomButton {...props} />
      </ThemeProvider>
    ).find(CustomButton);

    expect(typeof wrapper.prop("color")).toBe("string");
  });

  test("should have lineHeight prop as string data type", () => {
    const props = {
      color: "#FFFFFF",
    };
    const wrapper = mount(
      <ThemeProvider theme={lightTheme}>
        <CustomButton {...props} />
      </ThemeProvider>
    ).find(CustomButton);

    expect(typeof wrapper.prop("color")).toBe("string");
  });

  test("should have fontSize prop as string data type", () => {
    const props = {
      fontSize: "30rem",
    };
    const wrapper = mount(
      <ThemeProvider theme={lightTheme}>
        <CustomButton {...props} />
      </ThemeProvider>
    ).find(CustomButton);

    expect(typeof wrapper.prop("fontSize")).toBe("string");
  });

  test("should have borderRadius prop as string data type", () => {
    const props = {
      borderRadius: "0.8rem",
    };
    const wrapper = mount(
      <ThemeProvider theme={lightTheme}>
        <CustomButton {...props} />
      </ThemeProvider>
    ).find(CustomButton);

    expect(typeof wrapper.prop("borderRadius")).toBe("string");
  });
});
