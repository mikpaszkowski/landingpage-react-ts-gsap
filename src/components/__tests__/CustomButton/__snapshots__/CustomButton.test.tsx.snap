// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render correctly <CustomButton /> component 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.7rem 3.1rem;
  font-family: 'Reem Kufi',sans-serif;
  font-size: 2.5rem;
  font-style: normal;
  font-weight: normal;
  line-height: 3.6rem;
  border: none;
  border-radius: 8px;
  background-color: #FABD20;
  cursor: pointer;
  -webkit-transition: all 0.25s ease-in-out;
  transition: all 0.25s ease-in-out;
  overflow: hidden;
  will-change: transform;
}

.c0::after {
  content: "";
  position: absolute;
  z-index: -1;
  bottom: -100%;
  left: 0;
  height: 100%;
  width: 100%;
  border-radius: 8px;
  background-color: black;
  color: #FABD20;
  -webkit-transition: all 0.15s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: all 0.15s cubic-bezier(0.455,0.03,0.515,0.955);
}

.c0:hover {
  box-shadow: 0px 1rem 2rem rgba(0,0,0,0.25);
  color: #FABD20;
}

.c0:hover::after {
  -webkit-transform: translateY(-100%);
  -ms-transform: translateY(-100%);
  transform: translateY(-100%);
}

<button
  className="c0"
/>
`;
